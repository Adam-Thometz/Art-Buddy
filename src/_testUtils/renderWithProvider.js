import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';

// import renderWithRouter from './renderWithRouter';
import rootReducer from '../_redux/reducers/rootReducer';

function withProvider({ children, store }) {
  return (
    <Provider store={store}>
      <MemoryRouter>
        {children}
      </MemoryRouter>
    </Provider>
  )
};

function renderWithProvider(component, options) {
  const store = (options && options.store)
    ? options.store
    : rootReducer;
  return render(component, { wrapper: (args) =>
    withProvider({
      ...args,
      store
    }), ...options
  });
};

export default renderWithProvider;