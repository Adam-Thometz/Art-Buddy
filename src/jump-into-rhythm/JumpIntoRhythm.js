import React from 'react'

import WindowNavbar from '_components/window-nav/WindowNavbar';
import FrogLilyPads from './frog-lily-pad/FrogLilyPads';
import NoteDisplay from './note-display/NoteDisplay';
import NoteOptions from './note-options/NoteOptions';

const JumpIntoRhythm = () => {
  return (
    <>
      <WindowNavbar page='JUMP INTO RHYTHM' />
      <FrogLilyPads />
      <NoteDisplay />
      <NoteOptions />
    </>
  );
};

export default JumpIntoRhythm;