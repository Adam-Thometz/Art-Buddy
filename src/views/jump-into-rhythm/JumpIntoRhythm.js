import { useContext, useEffect } from "react";
import useVisited from "hooks/visited/useVisited";
import { PlayContext } from "context/PlayContext";

import { useSelector, useDispatch } from "react-redux";
import { clearGame } from "store/jump-into-rhythm/jumpIntoRhythmReducer";
import { changeCurrGame } from "store/_general/generalReducer";

import WindowNavbar from "components/window-nav/WindowNavbar";
import Instructions from "components/instructions/Instructions";
import Notes from "./notes/Notes";
import NoteOptions from "./note-options/NoteOptions";
import FrogLilyPad from "./frog-lily-pad/FrogLilyPads";

import makeFrog from "lib/jump-into-rhythm/makeFrog";
import { jumpIntoRhythm } from "data/_activities/activityList";
import { Transport } from "tone";

const JumpIntoRhythm = () => {
  const [hasVisited, setHasVisited] = useVisited(jumpIntoRhythm.lsKey);
  const { currGame } = useSelector((state) => state.general);
  const { isDisplayingLilyPads } = useSelector((state) => state.jumpIntoRhythm);
  const { setPlayFn } = useContext(PlayContext);
  const dispatch = useDispatch();

  useEffect(() => {
    dispatch(changeCurrGame(jumpIntoRhythm));
    setPlayFn(() => makeFrog().hop);
    Transport.bpm.value = 90;
    return () => {
      dispatch(clearGame());
      dispatch(changeCurrGame({}));
      Transport.stop();
      Transport.bpm.value = 120;
    };
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [dispatch]);

  return (
    <>
      <WindowNavbar page={currGame.name} />
      {!hasVisited ? (
        <Instructions setVisited={setHasVisited} />
      ) : (
        <>
          {isDisplayingLilyPads ? <FrogLilyPad /> : <Notes />}
          <NoteOptions />
        </>
      )}
    </>
  );
};

export default JumpIntoRhythm;
