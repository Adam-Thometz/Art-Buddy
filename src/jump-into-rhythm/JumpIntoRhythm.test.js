import React from "react";

import renderWithProvider from "_testUtils/renderWithProvider";

import JumpIntoRhythm from "./JumpIntoRhythm";

describe('JumpIntoRhythm component', () => {
  it('renders without crashing', () => {
    renderWithProvider(<JumpIntoRhythm />);
  });

  it('matches snapshot', () => {
    const { asFragment } = renderWithProvider(<JumpIntoRhythm />);
    expect(asFragment()).toMatchSnapshot();
  });
});