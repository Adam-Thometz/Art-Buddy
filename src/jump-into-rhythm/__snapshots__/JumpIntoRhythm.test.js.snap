// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JumpIntoRhythm component matches snapshot 1`] = `
<DocumentFragment>
  <main
    class="JumpIntoRhythm"
  >
    <header
      class="WindowNavbar-games"
    >
      <h1
        class="WindowNavbar-label"
      >
        JUMP INTO RHYTHM
      </h1>
      <div
        class="WindowNavbar-corner"
      >
        <img
          alt="Help"
          class="Help"
          src="file"
        />
      </div>
    </header>
    <header>
      <p>
        HULLO IM JUMP INTO RHYTHM. MAKE SOME RHYTHMS TO MAKE THE FROGGY JUMP PLZ
      </p>
    </header>
    <section
      class="FrogLilyPads"
    >
      <img
        alt="frog"
        class="Frog"
        src="file"
      />
      <div
        class="LilyPadDisplay"
      >
        <div
          class="LilyPad"
        />
        <div
          class="LilyPad"
        />
        <div
          class="LilyPad"
        />
        <div
          class="LilyPad"
        />
      </div>
    </section>
    <section
      class="NoteDisplay"
    >
      <img
        alt="Treble clef"
        src="file"
      />
      <section
        class="NoteDisplay-blocks"
      >
        <div
          class="NoteBlock"
          style="border-color: #fd4d3a;"
        />
        <div
          class="NoteBlock"
          style="border-color: #ffe81a;"
        />
        <div
          class="NoteBlock"
          style="border-color: #89eb0a;"
        />
        <div
          class="NoteBlock"
          style="border-color: #17255a;"
        />
      </section>
    </section>
    <section
      class="NoteOptions"
    >
      <section
        class="NoteOptions-notes"
      >
        <figure
          class="Icon"
          id="quarterNote"
        >
          <img
            alt=""
            src="file"
            style="height: 100px; width: 30%; cursor: pointer;"
          />
          <figcaption
            class="Icon-text"
            style="font-size: 1.75rem;"
          >
            Quarter Note
          </figcaption>
        </figure>
        <figure
          class="Icon"
          id="eighthNotes"
        >
          <img
            alt=""
            src="file"
            style="height: 100px; width: 30%; cursor: pointer;"
          />
          <figcaption
            class="Icon-text"
            style="font-size: 1.75rem;"
          >
            Eighth Notes
          </figcaption>
        </figure>
      </section>
      <button
        class="Button regular"
        style="border-color: #89eb0a; margin: 0px; height: 10vh;"
      >
        PLAY
      </button>
    </section>
  </main>
</DocumentFragment>
`;
